core/src/gl/renderState.cpp

```diff
@@ -21,4 +21,12 @@ namespace RenderState {
         glClearColor(0.3, 0.3, 0.3, 1.0);
     }
     
+    void invalidateAllStates() {
+        blending.invalidate();
+        culling.invalidate();
+        depthTest.invalidate();
+        depthWrite.invalidate();
+        blendingFunc.invalidate();
+    }
+    
```

Comment:
Yes this works, I didn't want to break the pattern followed so far for invalidation though (the gl object itself is aware of invalidation and regenerate/reinit itself on context destroy).